src/PAModel/Checksum/IHashMaker.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Xml.Linq;
+
+
+namespace Microsoft.PowerPlatform.Formulas.Tools
+{
+    // Helpers for creating chceksums. 
+    interface IHashMaker : IDisposable
+    {
+        void AppendStartObj();
+        void AppendPropName(string name);
+        void AppendEndObj();
+        void AppendStartArray();
+        void AppendEndArray();
+
+        void AppendData(string value);
+        void AppendData(double value);
+        void AppendData(bool value);
+        void AppendNull();
+
+        // Called after all Appends(). 
+        byte[] GetFinalValue();
+    }
+
+    // Create a checksum using an incremental hash.    
+    class Sha256HashMaker : IHashMaker, IDisposable
+    {
+        private readonly IncrementalHash _hash;
+        public Sha256HashMaker()
+        {
+            _hash = IncrementalHash.CreateHash(HashAlgorithmName.SHA256);
+        }
+
+        public void AppendData(double value)
+        {
+            _hash.AppendData(value);
+        }
+
+        public void AppendData(string value)
+        {
+            _hash.AppendData(value);
+        }
+
+        public void AppendData(bool value)
+        {
+            _hash.AppendData(value);
+        }
+
+        public void AppendPropName(string name)
+        {
+            this.AppendData(name);
+        }
+
+        public void AppendStartObj()
+        {
```

Comment:
Worth noting this are all empty in the real checksum. We may want to consider hashing something here.